x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -715,7 +715,7 @@ class CacheFileRegion extends EvictableRefCounted {
         CacheFileRegion(RegionKey<KeyType> regionKey, int regionSize) {
             this.regionKey = regionKey;
             assert regionSize > 0;
-            tracker = new SparseFileTracker("file", regionSize);
+            tracker = new SparseFileTracker(regionKey.toString(), regionSize);
```

Comment:
I think we used `file` here to avoid consuming a lot of heap for the description. I would prefer to revert this change, unless we have a good reason to do it?

Reply:
Thanks for explaining the subtlety. I have reverted this change and also added a comment about the choice of using string constant. 

Btw, we have other places such as the following that uses non-constant string. Is it a concern or not?

https://github.com/elastic/elasticsearch/blob/eeecdbf87b08ec03682dabe7def7272055119252/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java#L137

Reply:
Thanks. 

I haven't seen Cold nodes heap dumps with many heap consumed by those descriptions but it looks like we could save a bit of heap by reducing the description there too.

Reply:
I'll leave them for now since there has not been any issue to avoid any surprises. Thanks!